### PR TITLE
Introduce `reactive.prepareForReuse` for reusable components.

### PR DESCRIPTION
**Pull request by andersio**
~~The protocol is fragile though, because:~~
~~It won't build because:
https://github.com/ReactiveCocoa/ReactiveCocoa/pull/3210#pullrequestreview-4397112~~

~~Looking for workarounds.~~

### CODE REVIEW

**Comment by liscio** [comment]
@andersio did you forget to check in the xcodeproj for these changes?

File: ReactiveCocoa/UIKit/ReusableComponents.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import UIKit
+import ReactiveSwift
+import enum Result.NoError
+
+@objc public protocol Reusable: class {
+	func prepareForReuse()
```
Comment (NachoSoto) [inline ReactiveCocoa/UIKit/ReusableComponents.swift]: 👍 thank you!

File: ReactiveCocoaTests/AppKit/ReusableComponentsSpec.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+			expect(isTriggered) == false
+
+			cell.prepareForReuse()
+			expect(isTriggered) == true
+		}
+
+		if #available(macOS 10.11, *) {
+			describe("NSCollectionViewItem") {
+				let item = TestViewItem()
```
Comment (NachoSoto) [inline ReactiveCocoaTests/AppKit/ReusableComponentsSpec.swift]: These are not tests, without `it` no code here will run!

File: ReactiveCocoaTests/AppKit/ReusableComponentsSpec.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import Quick
+import Nimble
+import Result
+import ReactiveSwift
+import ReactiveCocoa
+import AppKit
+
+class ReusableComponentsSpec: QuickSpec {
+	override func spec() {
+		describe("NSTableCellView") {
```
Comment (NachoSoto) [inline ReactiveCocoaTests/AppKit/ReusableComponentsSpec.swift]: Ditto

File: ReactiveCocoaTests/UIKit/ReusableComponentsSpec.swift [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import Quick
+import Nimble
+import Result
+import ReactiveSwift
+import ReactiveCocoa
+import UIKit
+
+class ReusableComponentsSpec: QuickSpec {
+	override func spec() {
```
Comment (NachoSoto) [inline ReactiveCocoaTests/UIKit/ReusableComponentsSpec.swift]: Ditto

**Comment by andersio** [comment]
Double checked. Others are fine, including the new InterceptingSpec.

**Comment by andersio** [comment]
Double checked. Others are fine, including the new InterceptingSpec.

**Pull request merged**